src/qt/pivx/governancemodel.cpp

```diff
@@ -173,12 +180,19 @@ void GovernanceModel::pollGovernanceChanged()
 {
     if (!isTierTwoSync()) return;
 
+    int chainHeight = clientModel->getNumBlocks();
     // Try to broadcast any pending for confirmations proposal
     auto it = waitingPropsForConfirmations.begin();
     while (it != waitingPropsForConfirmations.end()) {
         if (!g_budgetman.AddProposal(*it)) {
             LogPrint(BCLog::QT, "Cannot broadcast budget proposal - %s", it->IsInvalidReason());
-            it++;
+            if (it->GetBlockStart() >= chainHeight) {
+                // Edge case, the proposal was never broadcasted before the next superblock, can be removed.
+                // future: notify the user about it.
+                it = waitingPropsForConfirmations.erase(it);
```

Comment:
Ok... this was fixed in debbed9b15d0ddbd6e381f0a419397d23f119392 (still... could be squashed here, and I'd rather use `<=` instead of `<`) but the other one, in `txLoaded`, was missed (the broadcast should be scheduled only if the block start has not passed yet).

Reply:
yeah, agree.